src/firefly/js/api/ApiHighlevelBuild.js

```diff
@@ -123,10 +127,12 @@ function buildTablePart(llApi) {
      */
 
     /**
-     * The general plotting function to plot a table.
+     *@function showTable
      * @param {string|HTMLDivElement} targetDiv to put the table in.
      * @param {Object} request         request object created from
      * @param {TblOptions} options     table options.
+     * @memeberof firefly
+     * @example fierfly.showTable
```

Comment:
Actually, it does see the type definition and generates Type Definitions
section correctly, but it does not recognize the namepath and does not
create link to the type definition. (See
https://github.com/jsdoc3/jsdoc/issues/969)

There are 3 solutions:

1. @param {module:firefly~TblOptions} table options

or

2. mark typedef @static and use
    @param {module:firefly.TblOptions} table options


3. inline object definition:

  @param {Object} TblOptions - table options
  @param {string}  TblOptions.tbl_group - the group this table belongs to.
Defaults to 'main'.
  @param {boolean} TblOptions.removable - true if this table can be removed
from view.  Defaults to true.
  @param {boolean} TblOptions.showUnits - defaults to false
  @param {boolean} TblOptions.showFilters - defaults to false
  @param {boolean} TblOptions.selectable - defaults to true
  @param {boolean} TblOptions.expandable - defaults to true
  @param {boolean} TblOptions.showToolbar - defaults to true
  @param {boolean} TblOptions.border - defaults to true

2016-08-15 17:33 GMT-07:00 loitly <notifications@github.com>:

> In src/firefly/js/api/ApiHighlevelBuild.js
> <https://github.com/Caltech-IPAC/firefly/pull/146#discussion_r74860064>:
>
> >       * @param {string|HTMLDivElement} targetDiv to put the table in.
> >       * @param {Object} request         request object created from
> >       * @param {TblOptions} options     table options.
> > +     * @memeberof firefly
> > +     * @example fierfly.showTable
>
> This is nice. It works great for the these API functions. Too bad it does
> not recognize TblOptions @typedef <https://github.com/typedef> which is
> declared just above it. We'll have to figure out how to pull these typefefs
> in as well.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Caltech-IPAC/firefly/pull/146/files/08dcb7053c54e3b5cc77622e41b8a057621e2121#r74860064>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALU3QJ1wXPSOaJnzTehLdGQnkVjxYzYoks5qgQVJgaJpZM4JjjVP>
> .
>